On generating with relationship, causes constraint violation exception

<!--
- Please follow the issue template below for bug reports.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-->

##### **Overview of the issue**

When generating entities such as:
```
Author (with unique name).
Book (with ManyToOne to Author, required)
```
with filtering enables, causes errors in testing on violating unique constraint (creates 2 authors with the same name).

test and line are:
```
com.generation.testing.web.rest.AuthorResourceIT > getAllAuthorsByBookIsEqualToSomething() FAILED
    org.springframework.dao.DataIntegrityViolationException at AuthorResourceIT.java:339
        Caused by: org.hibernate.exception.ConstraintViolationException at AuthorResourceIT.java:339
            Caused by: org.h2.jdbc.JdbcSQLIntegrityConstraintViolationException at AuthorResourceIT.java:339
```

##### **Motivation for or Use Case**

It is a reasonable relationship, should be generatable.

##### **Reproduce the error**

Generate the application as suggested and run the tests.

##### **Related issues**

##### **Suggest a Fix**

##### **JHipster Version(s)**

```
generation-testing@0.0.0 /Users/test/repos/generation-testing
└── generator-jhipster@6.1.2 

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.generation.testing"
    },
    "jhipsterVersion": "6.1.2",
    "applicationType": "monolith",
    "baseName": "generationTesting",
    "packageName": "com.generation.testing",
    "packageFolder": "com/generation/testing",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "hazelcast",
    "enableHibernateCache": false,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "mysql",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "gradle",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "bXktc2VjcmV0LXRva2VuLXRvLWNoYW5nZS1pbi1wcm9kdWN0aW9uLWFuZC10by1rZWVwLWluLWEtc2VjdXJlLXBsYWNl",
    "useSass": true,
    "clientPackageManager": "npm",
    "skipClient": true,
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "enableTranslation": false
  }
}
</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Author {
  name String required unique,
  birthday ZonedDateTime required
}
entity Book {
  name String required,
  publishedOn ZonedDateTime required
}
relationship OneToMany {
  Author{book} to Book{author(name) required}
}

paginate Author, Book with pagination
service Author, Book with serviceClass
filter Author, Book

</pre>
</details>


##### **Environment and Tools**

openjdk version "11.0.3" 2019-04-16
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.3+7)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.3+7, mixed mode)

git version 2.19.1

node: v8.11.4

npm: 6.4.1

yeoman: 2.0.1

yarn: 1.9.4

Docker version 18.09.2, build 6247962

docker-compose version 1.23.2, build 1110ad01

##### **Browsers and Operating System**

Irrelevant

-   [x] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

--- a/generators/entity-server/templates/src/test/java/package/web/rest/EntityResourceIT.java.ejs
+++ b/generators/entity-server/templates/src/test/java/package/web/rest/EntityResourceIT.java.ejs
@@ -888,6 +888,7 @@ _%>
         <%= asEntity(relationship.otherEntityNameCapitalized) %> <%= relationship.relationshipFieldName %> = <%= asEntity(entityInstance) %>.get<%= relationship.relationshipNameCapitalized %>();
         <%_ } else { _%>
         // Initialize the database
+        <%= entityInstance %>Repository.saveAndFlush(<%= asEntity(entityInstance) %>);
         <%= asEntity(relationship.otherEntityNameCapitalized) %> <%= relationship.relationshipFieldName %> = <%= relationship.otherEntityNameCapitalized %>ResourceIT.createEntity(em);
         em.persist(<%= relationship.relationshipFieldName %>);
         em.flush();

